src/libsync/propagatecommonzsync.h

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) by Ahmed Ammar <ahmed.a.ammar@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#pragma once
+
+#include <QLoggingCategory>
+#include <QTemporaryFile>
+
+extern "C" {
+    #include "librcksum/rcksum.h"
+    #include "libzsync/zmap.h"
+    #include "libzsync/sha1.h"
+    #include "libzsync/zsync.h"
+    #include "format_string.h"
+}
+
+namespace OCC {
+Q_DECLARE_LOGGING_CATEGORY(lcZsyncPut)
+Q_DECLARE_LOGGING_CATEGORY(lcZsyncGet)
+
+enum class ZsyncMode { download, upload };
+
+/**
+ * @ingroup libsync
+ *
+ * Function to know if we are allowed to attempt using zsync from configuration/command-line options.
+ *
+ */
+bool isZsyncPropagationEnabled(OwncloudPropagator *propagator, SyncFileItemPtr item);
+
+/**
+ * @ingroup libsync
+ *
+ * Propagation zsync seeding heleper job
+ *
+ */
+class ZsyncSeedJob : public QObject
+{
+    Q_OBJECT
+    QByteArray _zsyncData;
+    QString _zsyncFilePath;
+    ZsyncMode _type;
+public:
+    explicit ZsyncSeedJob(QByteArray &zsyncData, QString path, ZsyncMode type)
+        : _zsyncData(zsyncData)
+        , _zsyncFilePath(path)
+        , _type(type)
+    {
+    };
+
+    Q_INVOKABLE void seed();
+
+signals:
+    void finishedSignal(void *zs);
+    void failedSignal(QString errorString);
+};
+
+/**
+ * @ingroup libsync
+ *
+ * Propagation zsync file creation heleper job
+ *
```

Comment:
What does this job do? When is it used?

I haven't reviewed the implementation yet since it's hard without knowing what it's supposed to do.

Reply:
Fixed.